### PR TITLE
OCT-1495: Implement app-wide fiat support

### PR DESCRIPTION
**Pull request by jmikolajczyk**
## Definition of Done

1. [ ] Acceptance criteria are met.
2. [ ] PR is manually tested before the merge by developer(s).
    - [ ] Happy path is manually checked.
3. [ ] PR is manually tested by QA when their assistance is required (1).
    - [ ] Octant Areas & Test Cases are checked for impact and updated if required (2).
4. [ ] Unit tests are added unless there is a reason to omit them.
5. [ ] Automated tests are added when required.
6. [ ] The code is merged.
7. [ ] Tech documentation is added / updated, reviewed and approved (including mandatory approval by a code owner, should such exist for changed files).
    - [ ] BE: Swagger documentation is updated.
8. [ ] When required by QA:
    - [ ] Deployed to the relevant environment.
    - [ ] Passed system tests.

---

(1) Developer(s) in coordination with QA decide whether it's required. For small tickets introducing small changes QA assistance is most probably not required.

(2) [Octant Areas & Test Cases](https://docs.google.com/spreadsheets/d/1cRe6dxuKJV3a4ZskAwWEPvrFkQm6rEfyUCYwLTYw_Cc).

### CODE REVIEW

File: client/src/hooks/helpers/useGetValuesToDisplay.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  valueCrypto?: bigint;
+  valueString?: string;
+} & GetValueCryptoToDisplayProps;
+
+type GetValuesToDisplay = (props: GetValuesToDisplayProps) => {
+  primary: string;
+  secondary?: string;
+};
+
+export default function useGetValuesToDisplay(): GetValuesToDisplay {
```
Comment (aziolek) [inline client/src/hooks/helpers/useGetValuesToDisplay.ts:21]: Have you considered unit-testing this hook?

File: client/src/utils/getValueFiatToDisplay.ts [hunk 2]
```diff
@@ -14,15 +25,8 @@ export default function getValueFiatToDisplay({
...
-}: {
-  coinPricesServerDowntimeText?: DoubleValueProps['coinPricesServerDowntimeText'];
-  cryptoCurrency: DoubleValueProps['cryptoCurrency'];
-  cryptoValues?: Response;
-  displayCurrency: NonNullable<SettingsData['displayCurrency']>;
-  error?: any;
-  isUsingHairSpace?: boolean;
-  valueCrypto: DoubleValueProps['valueCrypto'];
-}): string {
+  showLessThanZero = false,
```
Comment (aziolek) [inline client/src/utils/getValueFiatToDisplay.ts:28]: `showLessThanOneCent` IMHO.

File: client/src/utils/getValueFiatToDisplay.ts [hunk 3]
```diff
@@ -56,5 +60,9 @@ export default function getValueFiatToDisplay({
     displayCurrency === 'jpy' ? 0 : 2,
   );
 
+  if (valueFiat === '0.00' && showLessThanZero) {
+    return `< ${prefix}0.00`;
```
Comment (aziolek) [inline client/src/utils/getValueFiatToDisplay.ts:64]: In JPY (Japanese Yen) there are no subdivisions like cents, so an exception needs to be handled here.

image

File: client/src/utils/getValueFiatToDisplay.ts [hunk 4]
```diff
@@ -47,7 +51,7 @@ export default function getValueFiatToDisplay({
     !cryptoValues[cryptoCurrency][displayCurrency] ||
     !valueCrypto
   ) {
-    return `${prefix}0.00`;
+    return showLessThanZero ? `< ${prefix}0.00` : `${prefix}0.00`;
```
Comment (aziolek) [inline client/src/utils/getValueFiatToDisplay.ts:54]: I think this ternary here is not required. It's for the loading states only. Or maybe I'm wrong?

**Review by aziolek — CHANGES_REQUESTED** [review]

File: client/src/hooks/helpers/useGetValuesToDisplay.ts (hunk 1 above)
Comment (jmikolajczyk) [reply client/src/hooks/helpers/useGetValuesToDisplay.ts:21]: I considered but i decided to drop it. This hook returns unmodified data from utils that are already tested (main logic is covered).

**Review by aziolek — DISMISSED** [review]

**Review by aziolek — CHANGES_REQUESTED** [review]
1. Suffixes in projects view are gone.
- Production:

image

- This branch:

image

2. Suffixes in project view are gone too, the same situation with rewards.

3. Everywhere in metrics there are no suffixes when fiat is the default currency. Examples:

image


image

**Comment by jmikolajczyk** [comment]
> 1. Suffixes in projects view are gone.
> 
> * Production:
> 
> image
> 
> * This branch:
> 
> image
> 
> 2. Suffixes in project view are gone too, the same situation with rewards.
> 3. Everywhere in metrics there are no suffixes when fiat is the default currency. Examples:
> 
> image
> 
> image

@aziolek fixed

**Review by aziolek — APPROVED** [review]

**Comment by jmikolajczyk** [comment]
New PR with squashed commits -> https://github.com/golemfoundation/octant/pull/257

**Pull request closed without merging**